consensus/integration/blockordelay_test.go

```diff
@@ -9,68 +6,49 @@ import (
 
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/model/messages"
+	"github.com/onflow/flow-go/network"
 )
 
 // This file includes functions to simulate network conditions.
 // The network conditions are simulated by defining whether a message sent to a receiver should be
 // blocked or delayed.
 
-// block all messages sent by or received by a list of denied nodes for the first N messages
-func blockNodesForFirstNMessages(n int, denyList ...*Node) BlockOrDelayFunc {
-	denyDict := make(map[flow.Identifier]*Node, len(denyList))
-	for _, n := range denyList {
-		denyDict[n.id.ID()] = n
+// blockNodesFirstMessages blocks n incoming messages to given nodes
+func blockNodesFirstMessages(n uint64, denyList ...*Node) BlockOrDelayFunc {
+	blackList := make(map[flow.Identifier]uint64, len(denyList))
+	for _, node := range denyList {
+		blackList[node.id.ID()] = n
 	}
-
-	sent, received := 0, 0
-
-	return func(channelID string, event interface{}, sender, receiver *Node) (bool, time.Duration) {
-		block, notBlock := true, false
-
-		switch m := event.(type) {
+	return func(channel network.Channel, event interface{}, sender, receiver *Node) (bool, time.Duration) {
+		// filter only consensus messages
+		switch event.(type) {
 		case *messages.BlockProposal:
 		case *messages.BlockVote:
 		case *messages.BlockResponse:
-			log := receiver.log.With().Int("blocks", len(m.Blocks)).Uint64("first", m.Blocks[0].Header.View).
-				Uint64("last", m.Blocks[len(m.Blocks)-1].Header.View).Logger()
-			log.Info().Msg("receives BlockResponse")
-		case *messages.SyncRequest:
-		case *messages.SyncResponse:
-		case *messages.RangeRequest:
-		case *messages.BatchRequest:
+		case *messages.TimeoutObject:
 		default:
-			return notBlock, 0
-		}
-
-		if _, ok := denyDict[sender.id.ID()]; ok {
-			if sent >= n {
-				return notBlock, 0
-			}
-			sent++
-			return block, 0
+			return false, 0
 		}
-		if _, ok := denyDict[receiver.id.ID()]; ok {
-			if received >= n {
-				return notBlock, 0
-			}
-			received++
-			return block, 0
+		count, ok := blackList[receiver.id.ID()]
+		if ok && count > 0 {
+			blackList[receiver.id.ID()] = count - 1
+			return true, 0
 		}
 		return false, 0
 	}
 }
 
 func blockReceiverMessagesByPercentage(percent int) BlockOrDelayFunc {
 	rand.Seed(time.Now().UnixNano())
-	return func(channelID string, event interface{}, sender, receiver *Node) (bool, time.Duration) {
+	return func(channel network.Channel, event interface{}, sender, receiver *Node) (bool, time.Duration) {
 		block := rand.Intn(100) <= percent
 		return block, 0
 	}
 }
 
 func delayReceiverMessagesByRange(low time.Duration, high time.Duration) BlockOrDelayFunc {
 	rand.Seed(time.Now().UnixNano())
```

Comment:
generally, it is considered statistically not sound to split the stream of random numbers from one pseudo-random number generator [prng] into independent sub-streams. For example, if you have a prng and you use values for purpose A and purpose B alternating:
* From the perspective of process A, you want a steam of numbers with high randomness and independent of the randomness for B. 
* But exactly that quality of "independence" is broken by sub-sampling both streams A and B from the same prng. Furthermore, a sub-sample of a random stream generally does not preserve the quality of randomness and should be avoided. 

These are relevant for statistically sound experiments. While the quality of randomness is not that important here, it is still beneficial to follow best practises. Specifically, the extra engineering effort would be marginal here.